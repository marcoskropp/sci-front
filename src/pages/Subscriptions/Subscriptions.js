import { Paper, Button, Container, Typography, } from "@material-ui/core";
import React, { useState, useEffect, forwardRef } from 'react';
import { Link } from 'react-router-dom';

import { index } from '../../services/Subscriptions.services';
import TableRow from '../../components/TableRow.js'

export default function Subscriptions() {
    const [workshops, setWorkshops] = useState([]);

    useEffect(() => {
        const getWorkshops = async () => {
            const myWorkshops = await index();

            setWorkshops(myWorkshops);
        };
        getWorkshops();
    }, []);

    const save = () => {

    }

    const link = forwardRef((props, ref) => (
        <Link innerRef={ref} {...props} />
    ));

    const renderTable = () => {
        return (
            <table>
                <thead>
                    <tr>
                        <th>Título</th>
                        <th>Descrição</th>
                        <th>Local</th>
                        <th>Data de Início</th>
                        <th>Data de Término</th>
                    </tr>
                </thead>
                <tbody>
                    {workshops.map(workshop => (
                        <TableRow key={workshop.id} {...workshop} />
                    ))}
                </tbody>
            </table>
        );
    };

    return (
        <Container>
            <Paper style={{ padding: "30px" }}>
                <Typography variant="h4" component="h2" gutterBottom>
                    INSCREVER-SE
                </Typography>
                {renderTable()}
                <Button
                    variant="contained"
                    color="secondary"
                    onClick={save()}
                >
                    Salvar
                </Button>
                <Button
                    variant="contained"
                    color="secondary"
                    to="/"
                    component={link}
                >
                    Voltar
                </Button>
            </Paper>
        </Container>
    );
}
