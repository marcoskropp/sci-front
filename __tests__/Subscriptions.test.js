import React from 'react';
import { shallow } from 'enzyme';

import Subscriptions from '../src/pages/Subscriptions/Subscriptions';

describe('Testing Subscriptions Page', () => {
  it('Snapshot renders', () => {
    const wrapper = shallow(<Subscriptions />);

    expect(wrapper).toMatchSnapshot();
  });
});
