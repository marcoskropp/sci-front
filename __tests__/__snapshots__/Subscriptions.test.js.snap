// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Subscriptions Page Snapshot renders 1`] = `ShallowWrapper {}`;
